argoverse/map_representation/map_api_v2.py

```diff
@@ -0,0 +1,462 @@
+# <Copyright 2021, Argo AI, LLC. Released under the MIT license.>
+
+"""
+API for loading and Argoverse 2.0 maps. These include left and right lane boundaries,
+instead of only lane centerlines, as was the case in Argoverse 1.0 and 1.1.
+
+Separate map data (files) is provided for each log/scenario. This local map data represents
+map entities that fall within some distance according to l-infinity norm from the trajectory
+of the egovehicle (AV).
+"""
+
+import copy
+import glob
+import logging
+import os
+from pathlib import Path
+from typing import Dict, List, Mapping, NamedTuple, Optional, Tuple, Union
+
+import numpy as np
+from dataclasses import dataclass
+
+from argoverse.utils.centerline_utils import convert_lane_boundaries3d_to_polygon3d
+from argoverse.utils.dilation_utils import dilate_by_l2
+from argoverse.utils.interpolate import interp_arc
+from argoverse.utils.json_utils import read_json_file
+from argoverse.utils.sim2 import Sim2
+
+_PathLike = Union[str, "os.PathLike[str]"]
+
+ROI_ISOCONTOUR = 5.0  # in meters
+
+logger = logging.getLogger(__name__)
+
+
+class PedCrossing(NamedTuple):
+    """
+    Args:
+        edge1: array of shape (N,3) representing
+        edge2: array of shape (N,3) representing
+    """
+    edge1: np.ndarray
+    edge2: np.ndarray
+
+    def get_edges(self) -> Tuple[np.ndarray, np.ndarray]:
+        """ """
+        return (self.edge1[:, :2], self.edge2[:, :2])
+
+    def __eq__(self, other: "PedCrossing") -> bool:
+        """Check if two pedestrian crossing objects are equal, up to a tolerance."""
+        return np.allclose(self.edge1, other.edge1) and np.allclose(self.edge2, other.edge2)
+
+
+class LocalLaneMarking(NamedTuple):
+    """Information about a lane marking, representing either the left or right boundary of a lane segment.
+
+    Args:
+        bnd_type: 
+        src_lane_id:
+        bnd_side: string representing which side of a lane segment this marking represents, i.e. "left" or "right".
+        polyline: array of shape (N,3)
+    """
+
+    bnd_type: str
+    src_lane_id: int
+    bnd_side: str
+    polyline: np.ndarray
+
+
+@dataclass(frozen=False)
+class VectorLaneSegment:
+    """
+    Args:
```

Comment:
hmm is this convention from a style guide?

Reply:
marking as resolved, since we use `Args:` for other dataclasses e.g. [here](https://github.com/argoai/argoverse-api/blob/master/argoverse/evaluation/detection/eval.py#L84) in argoverse-api, so we can continue to use `Args:` for consistency here.